pkg/controllers/manila/manila.go

```diff
@@ -103,6 +103,9 @@ func (c *ManilaController) sync(ctx context.Context, syncCtx factory.SyncContext
 		case *gophercloud.ErrEndpointNotFound:
 			// OpenStack does not support manila, report the operator as disabled
 			return c.setDisabledCondition("This OpenStack cluster does not provide Manila service")
+		case gophercloud.ErrDefault404:
+			// Manila API is modified and not supported by the operator
+			return c.setDisabledCondition("Manila API is modified and not supported by the operator")
```

Comment:
The thing is that [`GetShareTypes()`](https://github.com/openshift/csi-driver-manila-operator/blob/2d0d211/pkg/controllers/manila/openstack.go#L37) can return many more legitimate errors than `ErrDefault403` or `ErrEndpointNotFound`. As we can see from the logs, we're getting a 404 and so far we don't know what function call is causing it.

I think we need to:
- add more logs to `GetShareTypes()` to identify what function call returns a 404.
- improve the error message when we catch the 404: "Manila API is modified and not supported by the operator" is a supposition and not actual fact.

Think for example of a case where keystone is misconfigured and returns a wrong endpoint for Manila. What should be the error message?